builtin/common/serialize.lua

```diff
@@ -202,7 +207,7 @@ function core.deserialize(str, safe)
 	if not func then return nil, err end
 
 	-- math.huge is serialized to inf, NaNs are serialized to nan by Lua
-	local env = {inf = math_huge, nan = nan}
+	local env = {inf = math_huge, nan = nan, _ = {}}
```

Comment:
Doesn't this break forward compatibility?
(That is, older versions of `core.deserialize` won't work with newer outputs of `core.serialize`.)
Instead you could add `local _={};` to the start of the serialize output.

Actually, the same applies for `inf` and `nan` (forgotten in #11427).

Reply:
> Doesn't this break forward compatibility?

Do we require forward compatibility?

Reply:
Would be good to have at least.
Any bugs caused by this (that is by using the function incorrectly, e.g. by using a newer world in an older minetest version) are presumably super confusing and weird for users.

And without forward comp, you are basically changing the format, from lua code that has to be run to lua code that has to be run in a certain env. This also makes it impossible to change the format back in the future.

Reply:
Restoring forward compability then. I considered stuffing this into the environment (and keeping the serialize output concise) elegant but your points far outweigh that, and having this be plain Lua comes with an elegance of its own. I suppose I tried to hard to be different than the original serialize implementation when redoing it.

Reply:
Btw., this also raises the question whether inf and nan should also be changed back to not be in the env, like in pre-5.6.